### PR TITLE
Add DecimalFormat for formatting of the float values to string

### PR DESCRIPTION
**Pull request by tsybanov**
## Motivation (required)

Closes (https://github.com/facebook/fresco/issues/2504)
Details on the fix (https://github.com/facebook/fresco/issues/2504#issuecomment-657771489)

This PR fixes the local problem only. I believe it would be better to preformat float values in strings within `Flog.formatString()` by using `DecimalFormat`. However, in that case, it would increase complexity since it'd require placeholders parsing.

## Test Plan (required)

Since the logic hasn't changed, I believe using existing tests is enough. Due to the fact that the problem occurs on an unknown list of OS, I wasn't able to find an edge case to create a test.

### CODE REVIEW

File: imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import com.facebook.imagepipeline.common.RotationOptions;
 import com.facebook.imagepipeline.image.EncodedImage;
 import com.facebook.infer.annotation.Nullsafe;
+import java.text.DecimalFormat;
```
Comment (daddivan69) [inline imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java:18]:
```suggestion
import java.text.DecimalFormat;
```suggestion
import java.text.DecimalFormat;

```

```

**Review by daddivan69 — APPROVED** [review]

File: imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
 import com.facebook.imagepipeline.common.RotationOptions;
 import com.facebook.imagepipeline.image.EncodedImage;
```
Comment (daddivan69) [inline imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java:16]:
```suggestion
 import com.facebook.imagepipeline.image.EncodedImage;
```

File: imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java [hunk 3]
```diff
@@ -15,6 +15,7 @@
...
 import com.facebook.infer.annotation.Nullsafe;
+import java.text.DecimalFormat;
 import javax.annotation.Nullable;
 
 @Nullsafe(Nullsafe.Mode.STRICT)
```
Comment (daddivan69) [inline imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java:21]:
```suggestion
 @Nullsafe(Nullsafe.Mode.)
```

**Review by daddivan69 — APPROVED** [review]

File: imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java [hunk 4]
```diff
@@ -102,17 +103,22 @@ public static float determineDownsampleRatio(
     final float widthRatio = ((float) resizeOptions.width) / widthAfterRotation;
     final float heightRatio = ((float) resizeOptions.height) / heightAfterRotation;
     float ratio = Math.max(widthRatio, heightRatio);
+
+    // Format float values for the log message in advance
+    // due to possible String.Format() NPE within FLog.formatString() in some OS
+    final DecimalFormat dimensionsFormatter = new DecimalFormat("#.#");
```
Comment (daddivan69) [inline imagepipeline-base/src/main/java/com/facebook/imagepipeline/transcoder/DownsampleUtil.java:109]:
```suggestion
    final DecimalFormat dimensionsFormatter = new DecimalFormat("#.#");
```
